src/globus_cli/commands/collection/list.py

```diff
@@ -0,0 +1,115 @@
+from globus_cli.termio import FORMAT_TEXT_TABLE, formatted_print
+
+STANDARD_FIELDS = [
+    ("ID", "id"),
+    ("Display Name", "display_name"),
+    ("Owner", default_identity_id_resolver.field),
+    ("Collection Type", "collection_type"),
+    ("Storage Gateway ID", "storage_gateway_id"),
+]
+
+
+class ChoiceSlugified(click.Choice):
+    """
+    Allow either hyphens or underscores, e.g. both 'mapped-collections' or
+    'mapped_collections'
+    """
+
+    def convert(self, value, param, ctx):
+        if value is None:
+            return None
+        value = value.replace("_", "-")
+        return super().convert(value, param, ctx)
+
+
+def hyphen_to_underscore(ctx, param, value):
+    if not value:
+        return None
+    return [x.replace("-", "_") for x in value]
+
+
+@command("list", short_help="List all Collections on an Endpoint")
+@click.argument(
+    "endpoint_id",
+    metavar="ENDPOINT_ID",
+)
+@click.option(
+    "--filter",
+    "filters",
+    multiple=True,
+    # choices are shown with hyphens, but the command will receive them with underscores
+    type=ChoiceSlugified(
+        ["mapped-collections", "guest-collections", "managed-by-me", "created-by-me"]
+    ),
+    callback=hyphen_to_underscore,
+    help="""\
+Filter results to one of the specified categories of collections. Can be applied
+multiple times. Note that mutually exclusive filters are allowed and will find no
+results.
+
+The filters are as follows
+
+\b
+mapped-collections:
+  Only show collections with collection_type="mapped"
+
+\b
+guest-collections:
+  Only show collections with collection_type="guest"
+
+\b
+managed-by-me:
+  Only show collections where one of your identities has a role
+
+\b
+created-by-me:
+  Only show collections where one of your identities was the creator
+""",
+)
+@click.option(
+    "--include-private-policies",
+    is_flag=True,
+    help=(
+        "Include private policies. Requires administrator role on the endpoint. Some "
+        "policy data may only be visible in `--format JSON` output"
+    ),
+)
+@click.option(
+    "--mapped-collection-id",
+    default=None,
+    type=click.UUID,
+    help=(
+        "Filter results to Guest Collections on a specific Mapped Collection. This is "
+        "the ID of the Mapped Collection"
+    ),
+)
+@LoginManager.requires_login(
+    LoginManager.TRANSFER_RS, LoginManager.AUTH_RS, pass_manager=True
+)
+def collection_list(
+    login_manager,
+    *,
+    endpoint_id,
+    include_private_policies,
+    filters,
+    mapped_collection_id,
+):
+    """
+    List the Collections on a given Endpoint
+    """
+    client = get_gcs_client(endpoint_id)
+    params = {}
+    if mapped_collection_id:
+        params["mapped_collection_id"] = mapped_collection_id
+    if filters:
+        params["filter"] = ",".join(filters)
+    if include_private_policies:
+        params["include"] = "private_policies"
+    res = client.get_collection_list(**params)
```

Comment:
Of these parameters, [we only have `include` as a keyword arg in the SDK](https://globus-sdk-python.readthedocs.io/en/stable/services/gcs.html#globus_sdk.GCSClient.get_collection_list).

You'll have to put the others in `query_params`, and we can add support for them in SDK v3.1.

Reply:
I filed two tracking stories for the follow-up on this. We'll add in the SDK, and then update here after an SDK release. All slated for CLI v3.1